From 89745dacef789b7f85ebeb0a6a66f284e4a2b944 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:34 -0400
Subject: [tools] lib/traceevent: fix pointer-integer size mismatch

Message-id: <1408953210-26343-267-git-send-email-jolsa@redhat.com>
Patchwork-id: 88996
O-Subject: [PATCH RHEL7.1 BZ1133083 266/322] tools lib traceevent: fix pointer-integer size mismatch
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 0e9e79a13ab9d56b86db6538305babc23b1445cc
Author: Mark Rutland <mark.rutland@arm.com>
Date: Wed Jan 15 10:44:07 2014 +0000

description
===========
The scsi and cfg80211 plugins cast between unsigned long long and
pointers, which is problematic for architectures where unsigned long
long is wider than the native pointer size:

  linux/tools/lib/traceevent/plugin_scsi.c: In function 'process_scsi_trace_parse_cdb':
  linux/tools/lib/traceevent/plugin_scsi.c:408:26: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
    scsi_trace_parse_cdb(s, (unsigned char *) args[1], args[2]);

  linux/tools/lib/traceevent/plugin_cfg80211.c: In function 'process___le16_to_cpup':
  linux/tools/lib/traceevent/plugin_cfg80211.c:11:18: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
    uint16_t *val = (uint16_t *) args[0];

This patch adds an intermediate cast to unsigned long, silencing the
warning.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_cfg80211.c b/tools/lib/traceevent/plugin_cfg80211.c
index dcab8e8..57e9822 100644
--- a/tools/lib/traceevent/plugin_cfg80211.c
+++ b/tools/lib/traceevent/plugin_cfg80211.c
@@ -8,7 +8,7 @@ static unsigned long long
 process___le16_to_cpup(struct trace_seq *s,
          unsigned long long *args)
 {
- uint16_t *val = (uint16_t *) args[0];
+ uint16_t *val = (uint16_t *) (unsigned long) args[0];
  return val ? (long long) le16toh(*val) : 0;
 }
 
diff --git a/tools/lib/traceevent/plugin_scsi.c b/tools/lib/traceevent/plugin_scsi.c
index 6fb8e3e..7ef16cc 100644
--- a/tools/lib/traceevent/plugin_scsi.c
+++ b/tools/lib/traceevent/plugin_scsi.c
@@ -405,7 +405,7 @@ scsi_trace_parse_cdb(struct trace_seq *p, unsigned char *cdb, int len)
 unsigned long long process_scsi_trace_parse_cdb(struct trace_seq *s,
       unsigned long long *args)
 {
- scsi_trace_parse_cdb(s, (unsigned char *) args[1], args[2]);
+ scsi_trace_parse_cdb(s, (unsigned char *) (unsigned long) args[1], args[2]);
  return 0;
 }
 
-- 
1.7.1